app/src/main/java/com/cpr3663/cpr_scouting_app/PreMatch.java

```diff
@@ -217,33 +244,41 @@ public void onFocusChange(View view, boolean focus) {
                             Matches.MatchRow Match = Globals.MatchList.getMatchInfoRow(MatchNum);
                             if (Match != null) {
                                 // MUST CONVERT TO STRING or it crashes with out warning
-                                int[] Teams = Match.getListOfTeams();
-                                // TODO Set "Teams" to the options in the single select dropdown
+                            String[] teams_Str = new String[teams.length];
+                            for (int i = 0; i < teams.length; i++) {
+                                teams_Str[i] = String.valueOf(teams[i]);
+                            }
+                            if (teams_Str[0].equals("0")) {
+                                teams_Str = new String[]{"None"};
                             }
+                            ArrayAdapter<String> adp_Team = new ArrayAdapter<String>(view.getContext(), R.layout.cpr_spinner, teams_Str);
+                            adp_Team.setDropDownViewResource(R.layout.cpr_spinner_item);
+                            spinner_Team.setAdapter(adp_Team);
                         }
                     }
                 }
             }
         });
 
-        edit_Team.setOnFocusChangeListener(new View.OnFocusChangeListener() {
+        spinner_Team.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
             @Override
-            public void onFocusChange(View view, boolean focus) {
-                if (!focus) {
-                    String TeamToScoutStr = String.valueOf(edit_Team.getText());
-                    if (!TeamToScoutStr.isEmpty()) {
-                        int TeamToScout = Integer.parseInt(TeamToScoutStr);
-                        // Since TeamList always has a NO_TEAM entry at index=0, need to check size-1
-                        if (TeamToScout > 0 && TeamToScout < Globals.TeamList.size()) {
-                            // This will crash the app instead of returning null if you pass it an invalid num
-                            String ScoutingTeamName = Globals.TeamList.get(TeamToScout);
-                            text_TeamName.setText(ScoutingTeamName);
-                        } else {
-                            text_TeamName.setText("");
-                        }
+            public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l) {
+                String TeamToScoutStr = String.valueOf(spinner_Team.getSelectedItem());
+                if (!TeamToScoutStr.isEmpty() && !TeamToScoutStr.equals("None")) {
+                    int TeamToScout = Integer.parseInt(TeamToScoutStr);
+                    // Since TeamList always has a NO_TEAM entry at index=0, need to check size-1
+                    if (TeamToScout > 0 && TeamToScout < Globals.TeamList.size() - 1) {
```

Comment:
I think because you're checking <, you don't need to subtract 1 here.  While team 9999 is a test team number and likely won't be chosen...it might not show the name because we think it's too big.  (TeamList.size() should return 10000).